core/src/main/java/com/sequenceiq/cloudbreak/service/stack/CloudResourceAdvisor.java

```diff
@@ -199,6 +219,60 @@ public AutoscaleRecommendation getAutoscaleRecommendation(Long workspaceId, Stri
         return recommendAutoscale(blueprintTextProcessor);
     }
 
+    private Optional<String> getGroupNameFromClusterTemplate(String definitionName, Long workspaceId) {
+        Optional<String> groupResponse = Optional.empty();
+        if (!Strings.isNullOrEmpty(definitionName)) {
+            try {
+                groupResponse = transactionService.required(() -> {
+                    Optional<ClusterTemplate> templateByName = clusterTemplateService.getTemplateByName(definitionName, workspaceId);
+                    Optional<String> group = Optional.empty();
+                    if (templateByName.isPresent()) {
+                        if (templateByName.get().getStackTemplate() != null) {
+                            group = getGroupNameFromCustomTemplate(templateByName, group);
+                        } else {
+                            group = getGroupNameFromDefaultTemplate(definitionName, templateByName, group);
+                        }
+                    }
+                    return group;
+                });
+            } catch (Exception e) {
+                LOGGER.error("Could not parse Default Cluster with name {}. Error: {}", definitionName, e);
+            }
+        }
+        return groupResponse;
+    }
+
+    private Optional<String> getGroupNameFromDefaultTemplate(String definitionName, Optional<ClusterTemplate> templateByName, Optional<String> group) {
+        try {
+            DefaultClusterTemplateV4Request clusterTemplateV4Request = new Json(getTemplateString(templateByName.get()))
+                    .get(DefaultClusterTemplateV4Request.class);
+            group = clusterTemplateV4Request.getDistroXTemplate().getInstanceGroups()
+                    .stream()
+                    .filter(e -> InstanceGroupType.GATEWAY.equals(e.getType()))
+                    .map(e -> e.getName())
+                    .findFirst();
+        } catch (IOException e) {
+            LOGGER.error("Could not parse Default Cluster with name {}. Error: {}", definitionName, e);
+        }
+        return group;
+    }
+
+    private Optional<String> getGroupNameFromCustomTemplate(Optional<ClusterTemplate> templateByName, Optional<String> group) {
+        Optional<Stack> stackOptional = stackTemplateService.getByIdWithLists(templateByName.get().getStackTemplate().getId());
+        if (stackOptional.isPresent()) {
```

Comment:
Why do you need isPresent() and get() here?
group = stackOptional.map(Stack::getInstanceGroups)
                .map(Set::stream)
                .flatMap(igs -> igs.filter(e -> InstanceGroupType.GATEWAY.equals(e.getInstanceGroupType()))
                        .map(e -> e.getGroupName())
                        .findFirst())